libraries/Microsoft.Bot.Builder/Inspection/InspectionActivityExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder
+{
+    public static class InspectionActivityExtensions
+    {
+        private static JsonSerializerSettings jsonSerializerSettings = new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore };
+
+        public static IEnumerable<Activity> Clone(this List<Activity> activities)
+        {
```

Comment:
Why is the "this" argument a List<Activity>? Should it be "IEnumerable<Activity>" ?

Reply:
Yes you would have thought, however the usage is in the callback which is List<Activity> and polymorphic types are not understood by the extension method binding.

Anyhow this might go soon because we maybe moving everything to trace activities.